shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -149,7 +160,13 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
       // HashMap for FileWriter per each category
       Map<String, FileWriter> fileWriterMap = new HashMap<>();
       for (String fileName : fileNames) {
-        fileWriter = new FileWriter(PathUtils.concatPath(filePath, fileName));
+        if (validate) {
```

Comment:
this looks identical to the newly added section in `writeCSVFile`. refactor to a helper function?